### ISSUE TITLE
Could Alacritty start tagging versions?

### ISSUE THREAD

**Issue by anko** [jwilm/alacritty#1240]
Even if you don't feel "worthy" of a 1.0 for some reason, the reality is that people are using and packaging alacritty, and it would be reassuring if the commits packagers are checking in as the *current version of alacritty* into packaging systems aren't just their best guesses for okay cut points.

Even if the version is 0.x, would it be possible to start tagging versions occasionally?

My horse in this race is [getting the Void Linux alacritty package updated](https://github.com/voidlinux/void-packages/pull/11390), but I'm sure other distros' packagers would love to see versions tagged too.

**Comment by jwilm** [comment]
That sounds reasonable. I was going to do this when we start shipping binaries, but perhaps it's best not to block on that given that people are, as you say, packaging Alacritty in official repositories.

From your perspective as a packager, would it be helpful to add a tag to the current master as it is today? Naturally, future releases would be tagged as well.

**Comment by anko** [comment]
@jwilm If the current master head commit feels to you like something that could be a "version" (a relatively stable fixed point between feature branch merges, something people could and should be using as software), then yes—I'd love for that to be tagged. Or whatever is the latest such commit.

However, if you feel the project is still in so much flux that no commit stands out as more stable or version-cuttable than the others, then no—it wouldn't add any new information.

**Comment by chrisduerr** [comment]
I don't think adding versions to point out any kind of "stability" makes sense before 1.0. I would expect every commit that lands on master to be expected to work properly without introducing a bunch of regressions. At the same time I wouldn't expect any kind of "stability" guarantee from any of them.

Tagging versions is more about showing a difference I'd say. If there's just a bunch of commits which rework the config file/README/other stuff not directly impacting features, I would not tag versions because it doesn't add any direct improvement for the user.
However when new features are introduced that people are looking forward to (for example the scrollback branch which users are using even now before it's introduction into master), it should be indicated that there was a significant change which makes it "worth" to update the application.

So in summary, I'd stay away from promising any kind of "stability" and focus more on making clear when significant changes have been introduced.

I don't think it's hugely revelant if current master gets tagged or we wait until other features are added and start with future commits, as long as a consistent versioning scheme is started for maintainers.

**Comment by anko** [comment]
@chrisduerr I agree. That's what I meant by "relatively stable"; stable compared to other work-in-progress commits. It was probably a bad word choice. :sweat_smile: You explained it much better.

**Comment by vbrandl** [comment]
Maybe a bit OT but also not worth a separate issue:
If you are looking to ship binaries, I'd like to point out https://github.com/japaric/trust to automatically generate binaries for a variety of operating systems, architectures and libc implementations. I've used it myself (on a toy project) and it worked fine.

**Comment by jwilm** [comment]
@vbrandl that will be helpful for #80 / #1090  / #927 (just realized we have a bunch of dupes and tagged them all).

**Comment by davidak** [comment]
FYI: alacritty is currently packaged in 5 independent repositorys.

https://repology.org/metapackage/alacritty/versions

Rather stable 0.x releases would be great.

**Comment by FliiFe** [comment]
Versioning with tags would be of great help. I packaged alacritty on gentoo (the ebuild proposed in the readme is not working anymore), and it is kind of a pain to use live ebuilds to package it.
Tags would allow for easy versioning of packages across all distributions.

**Comment by clawoflight** [comment]
+1 from here. I'm not maintaining the Arch package, but I know from my own experience with other projects that it's really helpful to have tags.
Even if you only, say, bump 0.1.0 by ..1 when a group of features land.
That will also make versions comparable between distros and help with error reporting :)

**Comment by kpcyrd** [comment]
> FYI: alacritty is currently packaged in 5 independent repositorys.
> https://repology.org/metapackage/alacritty/versions

Note that most of those seem to package random commits instead of actual releases. There's currently no non `-git` package for alacritty in the archlinux user repository because that would require tags. I think if and how binaries should be shipped is a separate issue, as this is potentially more complex then just "every update that is pushed to crates.io should be tagged in git" :)

**Comment by chrisduerr** [comment]
crates.io will probably take longer than both tagging versions and providing github builds.

Based on the last time I looked into it, as soon as versions are tagged, it should be no problem to provide binaries through travis + github.

**Comment by mynameisfiber** [comment]
With #1329, we can get debian/osx builds on tags (and nightlies for that matter).
